src/ui/elements.tsx

```diff
@@ -93,3 +94,178 @@ export const Input = <I extends FieldValues, F>({
     </div>
   );
 };
+
+
+type SelectProps<I extends FieldValues, F> =
+  Pick<ReturnType<typeof useForm<I>>, 'register'> & {
+  /** Human readable string describing the field. */
+  label: string,
+  name: Path<I>,
+  /** Whether this field is required or may be empty. */
+  required: boolean,
+  /** Function validating the value and returning a string in the case of error. */
+  validate?: Validate<F, I>,
+  errors: Partial<Record<keyof I, FieldError>>,
+  /** Passed to the `<input>`. */
+  type?: HTMLInputTypeAttribute,
+  disabled?: boolean,
+  clearable?: boolean,
+  placeholder: string,
+  options?: F[],
+  defaultOptions: boolean,
+  loadOptions: (input: string, callback: (options: readonly F[]) => void) => void;
+  defaultValue?: F,
+  onChange: (newValue: SingleValue<F>, actionMeta: ActionMeta<F> ) => void | undefined,
+  value?: F,
+  control?: any,
+};
+
+
+// A styled `<Dropdown>`` element with a label. Displays errors and integrated with
+// `react-hook-form`.
+//
+// Props:
+// - `errors` and `register`: the two values returned by `useForm`
+// - `label`: the human readable string describing the field
+// - `name`, `type` and `rest`: passed as HTML attributes
+// - `required`: boolean, whether this field is required or may be empty
+// - `validate`: optional, a function validating the value and returning a
+//   string in the case of error.
+
+export const Dropdown = <I extends FieldValues, F> ({
```

Comment:
This should be a doc comment:

```
/**
 * fadfadf
 */
```

The prop descriptions should just be doc comments on the type above, as you partially already did.